pkg/services/ngalert/api/api_prometheus.go

```diff
@@ -291,6 +366,20 @@ func (srv PrometheusSrv) toRuleGroup(groupKey ngmodels.AlertRuleGroupKey, folder
 				newRule.Health = "error"
 			}
 
+			if len(withStates) > 0 {
+				if _, ok := withStates[alertState.State]; !ok {
+					continue
+				}
+			}
+
+			lbs := make(model.LabelSet)
```

Comment:
NIT: Surrounding in an empty check can save some unnecessary `LabelSet` creation if there are no `matchers`

Reply:
Done! 🙂